src/types.rs

```diff
@@ -42,7 +42,12 @@ pub fn rewrite_path(context: &RewriteContext,
 
     if let Some(ref qself) = qself {
         result.push('<');
-        result.push_str(&pprust::ty_to_string(&qself.ty));
+        let fmt_ty = if let Some(str) = qself.ty.rewrite(context, width, offset) {
```

Comment:
We're appending the fmt_ty to result, so shouldn't we at least push the original string in case we can't format this?